meinberlin/apps/actions/assets/timestamps.ts

```diff
@@ -0,0 +1,16 @@
+import { format } from 'timeago.js'
```

Comment:
somehow the type annotation there is weird to me.
I tried something and ended up with a fully typed version of this function. Maybe to much?

```
const formatTime: () => void = (): void => {
  const times: NodeListOf<HTMLTimeElement> =
    document.querySelectorAll('time.relative')

  times.forEach((element: HTMLTimeElement) => {
    const sevenDays: number = 60 * 60 * 24 * 7 * 1000
    const datetime: Date = new Date(element.dateTime)

    if (new Date().valueOf() - datetime.valueOf() < sevenDays) {
      element.textContent = format(datetime, 'de')
    }
  })
}
```

Reply:
i know, it is a bit contradictive to what I said about the typing of functions, but in these cases it is helpful.
I kind of like the typing of every single variable, for instance the `NodeListOf<>` and then the single element `HTMLTimeElement` is helpful imho..

what do you think?

Reply:
same, It's very helpful because it will warn you if an attribute doesn't exist for example. It just took me a bit to figure out what type I need

Reply:
@khamui 
`const formatTime: () => void = (): void => {`
I find this hard to read and I don't think declaring the function type here adds anything helpful. Adding the return type makes sense, because you need to know what the function has to return, knowing the function type on the other side at best tells us it's not a react component, but we already know that because it's returning void.

`const times: NodeListOf<HTMLTimeElement> = document.querySelectorAll('time.relative')`
Works for me as well, I do prefer my shorter version though.

`times.forEach((element: HTMLTimeElement) => {`
I prefer to leave out the type here as it is safely inferred and you can get the information by looking at the type of `times` easily.  But yeah, just personal preference

Reply:
haha yeah, I guess we will all have different preferences. I kind of agree with your `NodeListOf` argument, will change that!